source/cli-translate.js

```diff
@@ -0,0 +1,128 @@
+    selectCliLanguageNode.className += "btn text-dark bg-light mx-4";
+    selectCliLanguageNode.style.fontSize=".75rem";
+    selectCliLanguageNode.style.float="right";
+    selectCliLanguageNode.style.clear="both";
+    selectCliLanguageNode.style.position="relative";
+    selectCliLanguageNode.style.top="-18px";
+    selectCliLanguageNode.style.height="22px";
+    selectCliLanguageNode.style.padding="0px";
+    selectCliLanguageNode.onchange = function(){setCliChanged(this)};
+    selectCliLanguageNode.title = "Change Operating System Syntax";
+    var selectCliLanguageOptions = ["Linux","Windows","Bat File"];
+
+    for(let option of selectCliLanguageOptions){
+        var optionNode = document.createElement("option");
+        optionNode.text=option;
+        selectCliLanguageNode.appendChild(optionNode);
+    }
+    node.after(selectCliLanguageNode);
+    return selectCliLanguageNode;
+}
+
+const imagemagick_linux_to_windows_replacements = new Map([
+    [/\\\s*\n/gm, "^\n"], //replace newlines with ^\n
+    [/\s?\\(\(|\))\s?/gm, " $1 "], // unesecape parens, add space on either side to ensure compatibility "word\)" is valid in linux in windows "word)" is not.
+    [/\\!/gm,"!"], // unescape exclamation point.
+    [/'/gm, "\""], //single quote to double quote, breaks on nested quotes.
+    [/\&|\||\<|\>/gm, "\\$&"], //Escape &,|,<,> - note: appears unused.
+]);
+
+const imagemagick_linux_to_windows_bat_replacements = new Map([
+    [/%/gm, "%%"], //Double up % signs. Needed for bat files, but not for CMD.
+    [/^\s*#/gm, "::"], //Replace UNIX comment # with two colons. No support for inline comments.
+])
+
+function translateLinuxToWindows(node) {
+    imagemagick_linux_to_windows_replacements.forEach(function(value,key){
+        node.innerHTML = node.innerHTML.replaceAll(key,value);
+    });
+}
+
+function translateWindowsToBatch(node) {
+    imagemagick_linux_to_windows_bat_replacements.forEach(function(value,key){
+        node.innerHTML = node.innerHTML.replaceAll(key,value);
+    });
+}
+
+function addLinuxWindowsCLIToggle(node) {
+    var copyNode = node.cloneNode(true);
+    copyNode.classList.add("cli-windows");
+    copyNode.style.display = "none";
+    translateLinuxToWindows(copyNode);
+    var batchNode = copyNode.cloneNode(true);
+    batchNode.classList.add("cli-batch");
+    batchNode.style.display = "none";
+    translateWindowsToBatch(batchNode);
+    if(batchNode.innerHTML == node.innerHTML) // No change, ignore.
+    {
+        copyNode.remove();
+        return [];
+    }
+
+    node.classList.add("cli-linux");
+    node.after(copyNode);
+    copyNode.after(batchNode);
+    return [copyNode,batchNode];
+}
+
+function addCliTranslations() {
+    var pres = document.querySelectorAll('pre.cli');
+    var cliWindowsList = [];
+    var cliBatchList = [];
+    var cliLinuxList = [];
+    var cliSelectList = [];
+    for (let pre of pres) {
+        var samp = pre.childNodes[0];
```

Comment:
Could you not do this with a query selector instead?

Reply:
I probably could. I need the pre and the samp child though, because I addSelectOptions(pre) and I modify the samp, so I'm not sure there is any value in selecting the samp child with the query selector.

Maybe with the query selector I could select only pre.cli that have a samp child, to get a smaller list to iterate over.

Reply:
Actually, I guess I could do:
```
var pres = document.querySelectorAll('pre.cli');
for( let pre of pres){
    var samp = pre.querySelector("samp");
    if( samp != null ) {
```

Which may have been what you meant. 👍

Reply:
Yup that is what I meant.